### DISCUSSION TITLE
Default columns?

### DISCUSSION THREAD

**Discussion by danbolinson-bt** [anvilistas/tabulator#49] [Ideas]
Tabulator does not work when you simply set the data attribute to a list of dicts - the table doesn't populate until you've set the columns property as well.

For mock-up purposes (and assuming a list of dicts) I'm often doing this:
```
self.tabulator_1.data = self.records
self.tabulator_1.columns = [{'title': k, 'field':k} 
                            for k in self.records[0]] # we could also do dict(self.records[0]) to work for datatable rows
```

What do you think of making that the default behavior? i.e. when you set data the columns are evaluated from the first row?

**Comment by s-cork** [comment]
@danbolinson-bt  Have you tried setting auto columns to True?

**Reply by danbolinson-bt** [reply]
Nope, that's what I was missing! Thanks.

**Reply by s-cork** [reply]
You can either set it with `self.tabulator.options` or in the designer view.
